nautobot/dcim/migrations/0030_migrate_region_and_site_data_to_locations.py

```diff
@@ -149,322 +170,321 @@ def create_site_location_type_locations(
-        dst_relationship_associations.update(destination_type=location_ct)
-        for relationship_association in dst_relationship_associations:
-            dst_region = Region.objects.get(id=relationship_association.destination_id)
-            dst_loc = Location.objects.get(name=dst_region.name, location_type=region_lt)
-            relationship_association.destination = dst_loc
-            relationship_association.destination_id = dst_loc.pk
-        RelationshipAssociation.objects.bulk_update(dst_relationship_associations, ["destination_id"], 1000)
-        dst_relationships = Relationship.objects.filter(destination_type=region_ct)
-        dst_relationships.update(destination_type=location_ct)
-
-        WebHook = apps.get_model("extras", "webhook")
-        web_hooks = WebHook.objects.filter(content_types__in=[region_ct])
-        for wh in web_hooks:
-            wh.content_types.add(location_ct)
-
-    # Reassign Site Models to Locations of Site LocationType
-    if Site.objects.exists():  # Iff Site instances exist
+    model_class = apps.get_model("dcim", f"{model}")
+    model_ct = ContentType.objects.get_for_model(model_class)
+    if model == "region":
+        model_lt = LocationType.objects.get(name="Region")
+    else:
+        model_lt = LocationType.objects.get(name="Site")
+
+        # Only Site Related models
         CircuitTermination = apps.get_model("circuits", "circuittermination")
         Device = apps.get_model("dcim", "device")
         PowerPanel = apps.get_model("dcim", "powerpanel")
         RackGroup = apps.get_model("dcim", "rackgroup")
         Rack = apps.get_model("dcim", "rack")
-        ComputedField = apps.get_model("extras", "computedfield")
-        ConfigContext = apps.get_model("extras", "configcontext")
-        CustomField = apps.get_model("extras", "customfield")
         CustomLink = apps.get_model("extras", "customlink")
-        DynamicGroup = apps.get_model("extras", "DynamicGroup")
-        ExportTemplate = apps.get_model("extras", "exporttemplate")
         ImageAttachment = apps.get_model("extras", "imageattachment")
-        JobHook = apps.get_model("extras", "jobhook")
-        Note = apps.get_model("extras", "note")
-        WebHook = apps.get_model("extras", "webhook")
-        Relationship = apps.get_model("extras", "relationship")
-        RelationshipAssociation = apps.get_model("extras", "relationshipassociation")
         Prefix = apps.get_model("ipam", "prefix")
         VLANGroup = apps.get_model("ipam", "vlangroup")
         VLAN = apps.get_model("ipam", "vlan")
         Cluster = apps.get_model("virtualization", "cluster")
 
-        site_lt.content_types.set(ContentType.objects.filter(FeatureQuery("locations").get_query()))
+        model_lt.content_types.set(ContentType.objects.filter(FeatureQuery("locations").get_query()))
 
         # Circuits App
         cts = CircuitTermination.objects.filter(location__isnull=True).select_related("site")
```

Comment:
On all of these querysets, can we `select_related("site__migrated_location")` for better efficiency?